File: guides/spring-boot-micronaut-data/java/src/main/java/example.micronaut/Book.java
package example.micronaut;

import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;

@MappedEntity
public record Book(
        @GeneratedValue
        @Id
        Long id,
        String title,
        int pages
) {
}

File: guides/micronaut-metrics/python/src/example/micronaut/book.py
from dataclasses import dataclass
from typing import Annotated

from micronaut.data.annotation import GeneratedValue, Id, MappedEntity
from micronaut.serde.annotation import Serdeable


@dataclass
@Serdeable
@MappedEntity  # <1>
class Book:
    isbn: str
    name: str
    id: Annotated[int | None, Id, GeneratedValue] = None  # <2> <3>
